packing.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#ifdef __GNUC__
+#define PACK( __Declaration__ ) __Declaration__ __attribute__((__packed__))
+#endif
+
+#ifdef _MSC_VER
+#define PACK( __Declaration__ ) __pragma( pack(push, 1) ) __Declaration__ __pragma( pack(pop))
+#endif
```

Comment:
I'm not sure how to feel about this. GCC supports MSVC packing pragma so we could avoid doing this, but I'm not sure about clang.